### PR TITLE
Archival Object request bug

### PR DESCRIPTION
**Pull request by akofa-a**
check to see that the archival object has a top container reference before showing the 'add to request list' button

### CODE REVIEW

File: public/models/harvard_aeon/archival_object_mapper.rb [hunk 1]
```diff
@@ -8,6 +8,20 @@ def map_extensions(mapped, item, repository, resource, resource_json)
       mapped.ext(:level).name = item['level'].capitalize
     end
 
+    def request_permitted?(item)
+      Rails.logger.error(item['json']['jsonmodel_type'])
```
Comment (jessemartinez) [inline public/models/harvard_aeon/archival_object_mapper.rb]: remove this debug line

File: public/models/harvard_aeon/archival_object_mapper.rb [hunk 2]
```diff
@@ -8,6 +8,20 @@ def map_extensions(mapped, item, repository, resource, resource_json)
...
+      if item['json']['jsonmodel_type'] == "archival_object" and item['json']['instances'] and item['json']['instances'].length >= 1
+        item['json']['instances'].each do |my_instance|
+          if my_instance['instance_type'] and my_instance['instance_type'] == "text" and my_instance['subcontainer'] and my_instance['subcontainer']['topcontainer'] and my_instance['subcontainter']['topcontainer']['ref']
+            true
```
Comment (jessemartinez) [inline public/models/harvard_aeon/archival_object_mapper.rb]: you should explicitly return true to break out of the loop and the method
```suggestion
           return  true
```

File: public/models/harvard_aeon/archival_object_mapper.rb [hunk 3]
```diff
@@ -8,6 +8,20 @@ def map_extensions(mapped, item, repository, resource, resource_json)
...
+          else
+            false
+          end
+        end
+      else
+        false
+      end
```
Comment (jessemartinez) [inline public/models/harvard_aeon/archival_object_mapper.rb]: it makes more sense to return false outside the top-most if block so it catches everything
```suggestion
      end 
      return false
```

File: public/models/harvard_aeon/archival_object_mapper.rb [hunk 4]
```diff
@@ -8,6 +8,20 @@ def map_extensions(mapped, item, repository, resource, resource_json)
...
     end
 
+    def request_permitted?(item)
+      Rails.logger.error(item['json']['jsonmodel_type'])
+      if item['json']['jsonmodel_type'] == "archival_object" and item['json']['instances'] and item['json']['instances'].length >= 1
+        item['json']['instances'].each do |my_instance|
+          if my_instance['instance_type'] and my_instance['instance_type'] == "text" and my_instance['subcontainer'] and my_instance['subcontainer']['topcontainer'] and my_instance['subcontainter']['topcontainer']['ref']
+            true
+          else
+            false
```
Comment (jessemartinez) [inline public/models/harvard_aeon/archival_object_mapper.rb]: no need to return false inside the loop; if the first loop fails then we want to check each `my_instance` value to see if it matches

**Review by jessemartinez — CHANGES_REQUESTED** [review]
Here's some feedback on the PR code.

**Comment by akofa-a** [comment]
@jessemartinez I've made the suggested changes! Thanks!

File: public/models/harvard_aeon/archival_object_mapper.rb [hunk 5]
```diff
@@ -9,18 +9,14 @@ def map_extensions(mapped, item, repository, resource, resource_json)
     end
 
     def request_permitted?(item)
-      Rails.logger.error(item['json']['jsonmodel_type'])
       if item['json']['jsonmodel_type'] == "archival_object" and item['json']['instances'] and item['json']['instances'].length >= 1
         item['json']['instances'].each do |my_instance|
           if my_instance['instance_type'] and my_instance['instance_type'] == "text" and my_instance['subcontainer'] and my_instance['subcontainer']['topcontainer'] and my_instance['subcontainter']['topcontainer']['ref']
```
Comment (jessemartinez) [inline public/models/harvard_aeon/archival_object_mapper.rb]: This is full of misspelled words. It should be `sub_container` and `top_container`

File: public/models/harvard_aeon/archival_object_mapper.rb [hunk 6]
```diff
@@ -9,18 +9,14 @@ def map_extensions(mapped, item, repository, resource, resource_json)
...
-          else
-            false
+            return true
           end
         end
-      else
-        false
-      end  
+      end
+      return false;
```
Comment (jessemartinez) [inline public/models/harvard_aeon/archival_object_mapper.rb]: Syntax error. No need for semicolon ;-)

**Review by jessemartinez — CHANGES_REQUESTED** [review]
A few more corrections

File: public/models/harvard_aeon/archival_object_mapper.rb (hunk 6 above)
Comment (jessemartinez) [reply public/models/harvard_aeon/archival_object_mapper.rb]: The semicolon still needs to be removed

**Comment by jessemartinez** [comment]
looks good. just make sure it works on the server!

**Pull request merged**